src/Xamarin.Android.Build.Tasks/Tasks/FindLayoutsToBind.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	public class FindLayoutsToBind : Task
+	{
+		static readonly Regex layoutPathRegex = new Regex ($".*{Path.DirectorySeparatorChar}layout(-?\\w+)*{Path.DirectorySeparatorChar}.*\\.a?xml$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+		public bool GenerateLayoutBindings { get; set; }
+
+		public string BindingDependenciesCacheFile { get; set; }
+
+		public ITaskItem[] BoundLayouts { get; set; }
+
+		[Required]
+		public ITaskItem[] ResourceFiles { get; set; }
+
+		[Output]
+		public ITaskItem[] LayoutsToBind { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogDebugMessage ("FindLayoutsToBind Task");
+			Log.LogDebugMessage ($"  GenerateLayoutBindings: {GenerateLayoutBindings}");
+			Log.LogDebugMessage ($"  BindingDependenciesCacheFile: {BindingDependenciesCacheFile}");
+			Log.LogDebugTaskItems ("  BoundLayouts:", BoundLayouts);
+			Log.LogDebugTaskItems ("  ResourceFiles:", ResourceFiles);
+
+			var layouts = new Dictionary <string, ITaskItem> (StringComparer.Ordinal);
```

Comment:
Collections & filesystem paths always concern me, because Windows & macOS are case-insensitive filesystems. This is why elsewhere we use `StringComparer.OrdinalIgnoreCase`. (In fact, and I guess this is my bad, the *only* place where `StringComparer.Ordinal` and *not* `StringComparer.OrdinalIgnoreCase` is used is in `GenerateLayoutCodeBehind.cs`!)

This should *probably* use `StringComparer.OrdinalIgnoreCase`.

Reply:
On one hand you're right, on the other - the file names are read by msbuild once and they shouldn't change during build time, and `Ordinal` is faster